stdlib/mexpr/symbolize.mc

```diff
@@ -411,7 +411,7 @@ lang ConAppTypeSym = ConTypeAst + AppTypeAst + AliasTypeAst + VariantTypeAst +
             appTy
         else
           if env.strictTypeVars then
-            if env.allowFree then TyCon t
+            if env.allowFree then smap_Type_Type (symbolizeType env) ty
```

Comment:
I think this could be `mkAppTy (TyCon t) args` (which would be more efficient since the type arguments are symbolized once already above).

This bug must have been introduced by me in #667, I'm sorry about that.

Reply:
Nice catch! I updated the code as you proposed.